packages/node/src/node.ts

```diff
@@ -1 +1,113 @@
-export default class Node {}
+import * as cf from "@counterfactual/cf.js";
+import { EventEmitter } from "events";
+import * as _ from "lodash";
+
+import { MultisigAddress, StateChannelInfo } from "./channel";
+
+export interface NodeOptions {
+  channels?: Map<MultisigAddress, StateChannelInfo>;
+}
+
+/**
+ * The Node class is the entry point to the @counterfactual/node package.
+ * It's responsible for:
+ * - encapsulating channels and their apps
+ * - connecting with a provided database to persistently store commitments
+ * - registering a signing service
+ */
+export class Node {
+  private channels: Map<MultisigAddress, StateChannelInfo> = new Map();
+
+  // Maps AppInstanceIDs to the EventEmitters sending/receiving events for
+  // the relevant AppInstance.
+  private eventEmitters: Map<string, EventEmitter> = new Map();
+
+  constructor(options: NodeOptions) {}
+
+  /**
+   * Given a channel, this adds said channel to the state of the Node if a
+   * channel with the same multisig address does not exist.
+   * @param channel The channel to keep track of.
+   */
+  openChannel(channel: StateChannelInfo): [boolean, string] {
+    if (this.channels.has(channel.multisigAddress)) {
+      return [
+        false,
+        `Channel with multisig address ${
+          channel.multisigAddress
+        } is already open`
+      ];
+    }
+    this.channels.set(channel.multisigAddress, channel);
+    return [true, ""];
+  }
+
+  /**
+   * In the context of the Node, closing a channel means that the multisig's
+   * funds have been removed, thus making the multisig useless. Hence, its entry
+   * in this node's channels is erased.
+   * @param multisigAddress
+   */
+  closeChannel(multisigAddress: string): [boolean, string] {
+    if (this.channels.has(multisigAddress)) {
+      this.channels.delete(multisigAddress);
+      return [true, ""];
+    }
+    return [
+      false,
+      `Channel with multisig address ${multisigAddress} does not exist on this Node`
+    ];
+  }
+
+  /**
+   * Retrieve the list of channels that are open in this Node.
+   * TODO: add auth token to retrieve only authorized channel(s)
+   */
+  getChannels(): Map<MultisigAddress, StateChannelInfo> {
+    return this.channels;
+  }
+
+  /**
+   * Returns all of the apps installed across all of the channels in the Node.
+   */
+  getApps(): cf.legacy.app.AppInstanceInfo[] {
```

Comment:
if we can avoid re-using these legacy types I think  it will help us

Reply:
@mvanderh if/when they're available, lmk - I'll update this asap

Reply:
You can now take https://github.com/counterfactual/monorepo/blob/master/packages/cf.js/API_REFERENCE.md#data-type-appinstanceinfo and turn it into a Typescript interface & use it.

Reply:
sounds good

Reply:
Also, how is this supposed to be called? I see nowhere outside of tests that this is called.

I'm assuming you're gonna listen for a "getAppInstances" event on app instance event emitters.

Reply:
correct

Reply:
@cf19drofxots 

1. Yeah, not sure where the Node protocol data types should live. My intuition says CFjs since that's where the spec is located. 

2. I'm confused. I think you're wrong and that we _do_ want to call getAppInstances through the Node/NodeProvider EventEmitter. 

Also I don't get why you call it an "AppInstance EventEmitter" because in my mind a Node shouldn't have to have a concept of a single AppInstance. It only needs a concept of "something that wants to communicate with the Node".